src/main/java/org/galatea/starter/service/PriceFinderService.java

```diff
@@ -0,0 +1,60 @@
+package org.galatea.starter.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.galatea.starter.domain.AVDay;
+import org.galatea.starter.domain.AVStock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+
+/**
+ * A layer for transformation, aggregation, and business required when retrieving data from AlphaVantage.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PriceFinderService implements PriceFinderClient{
+
+  final static String AVQueryURL = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&";
+  final static String AVApikey = "XNRUM1DDXGFTDL82";
+
+  @Autowired
+  private ObjectMapper objectMapper;
+  @Autowired
+  private RestTemplate restTemplate;
+
+/**
+ * Get all available data regarding requested stock from AlphaVantage and requested N days
+ *
+ * @return
+ */
+  public ResponseEntity<AVStock> processText(final String text, int n) {
+    String AVGetRequest = (new StringBuilder()).append(AVQueryURL).append("symbol=")
+        .append(text).append("&apikey=").append(AVApikey).toString();
+    ResponseEntity<AVStock> data = restTemplate.getForEntity(AVGetRequest, AVStock.class);
+    data = data.ok(getNDays(data.getBody(), n));
+    ResponseEntity dataDesiredFormat = new ResponseEntity(data.getBody().getAVTimeSeries(), data.getStatusCode());
```

Comment:
These two lines are a bit confusing to read.
41 creates a new response entity (using the old response entity for some reason?) calling getNDays on the body of the current response entity? Maybe this should be split into a few lines, one where you trim down the # of days using ```getNDays``` and one where you create a new ```ResponseEntity<AVStock>``` using ```ResponseEntity.ok(dataForNDays.getAvTimeSeries())```